docs/getting-started/sdk/password-manager/web/interoperability.md

```diff
@@ -0,0 +1,269 @@
+
+Use `wasm-bindgen` when you need JavaScript to call into Rust directly or work with live Rust
+objects (like a struct or a function in Rust). Otherwise, default to `tsify` for easier handling of
+complex types and data exchange.
+
+## How to use `wasm-bindgen`
+
+### Basic functions
+
+To create functions that can be called from JavaScript using `wasm-bindgen`, you need to apply the
+`wasm_bindgen` attribute to the function and ensure the return type is something that can be
+converted into a JavaScript type. For instance, simple types like `u32`, `f64`, `bool`, and `String`
+can be returned directly:
+
+```rust
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub fn do_something() -> u32 {
+    42
+}
+
+#[wasm_bindgen]
+pub fn say_hello() -> String {
+    "Hello, World!".to_owned()
+}
+```
+
+This will generate the following TypeScript definitions:
+
+```typescript
+/**
+ * @returns {number}
+ */
+export function do_something(): number;
+
+/**
+ * @returns {string}
+ */
+export function say_hello(): string;
+```
+
+### Structs
+
+If you want to return more complex types, you can annotate a Rust struct with the `wasm_bindgen`
+attribute. Make sure that any fields you want to access from JavaScript are public:
+
+```rust
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen]
+pub struct User {
+    pub age: u32,
+}
+```
+
+Which generates the following TypeScript definitions:
+
+```typescript
+export class User {
+  free(): void;
+  age: number;
+}
+```
+
+For structs containing `String` or other types that are not `Copy`, use the `getter_with_clone`
+attribute to ensure JavaScript can access the value without directly referencing it in WebAssembly
+memory:
+
+```rust
+#[wasm_bindgen(getter_with_clone)]
+pub struct User {
+    pub name: String,
+    pub age: u32,
+}
+```
+
+Which generates the following TypeScript definitions:
+
+```typescript
+export class User {
+  free(): void;
+  age: number;
+  name: string;
+}
+```
+
+This ensures that values like `String` are properly cloned before being passed to JavaScript.
+
+### Structs with Functions
+
+You can also return types with methods:
+
+```rust
+#[wasm_bindgen(getter_with_clone)]
+pub struct User {
+    pub name: String,
+    pub age: u32,
+}
+
+#[wasm_bindgen]
+impl User {
+    pub fn say_hello(&self) -> String {
+        format!("Hello, {}!", self.name)
+    }
+}
+```
+
+Which generates the following TypeScript definitions:
+
+```typescript
+export class User {
+  free(): void;
+  say_hello(): string;
+  age: number;
+  name: string;
+}
+```
+
+## How to use `tsify`
+
+While `wasm-bindgen` is ideal for exposing Rust functions and objects to JavaScript, `tsify` is
+better suited for exposing complex data models. The `tsify` approach generates TypeScript bindings
+automatically from Rust structs and works by serializing Rust data into JavaScript-compatible
+objects.
```

Comment:
I think we could remove the first clause in the first sentence here talking about `wasm-bindgen`, we already mention its strength elsewhere in the document, I don't think we need to mention it again in the `tsify` section.